src/apps/Altinn.AccessManagement/src/Altinn.AccessMgmt.Persistence/Services/AssignmentService.cs

```diff
@@ -139,50 +141,80 @@ public Task<bool> AddResourceToAssignment(Guid userId, Guid assignmentId, Guid r
         throw new NotImplementedException();
     }
 
-    [DoesNotReturn]
-    private static void Unreachable()
-    {
-        throw new UnreachableException();
-    }
-
     /// <inheritdoc/>
-    public async Task<Result<Assignment>> GetOrCreateAssignment2(Guid fromEntityId, Guid toEntityId, string roleCode, ChangeRequestOptions options, CancellationToken cancellationToken = default)
+    public async Task<ProblemInstance> DeleteAssignment(Guid fromEntityId, Guid toEntityId, string roleCode, ChangeRequestOptions options, bool cascade = false, CancellationToken cancellationToken = default)
     {
         ValidationErrorBuilder errors = default;
         var fromEntityExt = await entityRepository.GetExtended(fromEntityId, cancellationToken: cancellationToken);
-        if (fromEntityExt is null)
+        var toEntityExt = await entityRepository.GetExtended(toEntityId, cancellationToken: cancellationToken);
+        ValidatePartyIsNotNull(fromEntityId, fromEntityExt, ref errors, "$QUERY/party");
+        ValidatePartyIsOrg(fromEntityExt, ref errors, "$QUERY/party");
+        ValidatePartyIsNotNull(toEntityId, toEntityExt, ref errors, "$QUERY/to");
+        ValidatePartyIsOrg(toEntityExt, ref errors, "$QUERY/to");
+
+        var roleResult = await roleRepository.Get(t => t.Name, roleCode, cancellationToken: cancellationToken);
+        if (roleResult == null || !roleResult.Any())
+        {
+            Unreachable();
+        }
+
+        var roleId = roleResult.First().Id;
+        var existingAssignment = await GetAssignment(fromEntityId, toEntityId, roleId, cancellationToken);
+        if (existingAssignment == null)
         {
-            errors.Add(ValidationErrors.MissingPartyInDb, "/$QUERY/party", [new("partyId", fromEntityId.ToString())]);
+            return null;
         }
         else
         {
-            if (!fromEntityExt.Type.Name.Equals("Organisasjon"))
+            if (!cascade)
             {
-                errors.Add(ValidationErrors.InvalidPartyType, "/$QUERY/party", [new("partyId", $"expected party of type 'Organisasjon' got '{fromEntityExt.Type.Name}'.")]);
+                var packages = await assignmentPackageRepository.Get(f => f.AssignmentId, existingAssignment.Id, cancellationToken: cancellationToken);
+                if (packages != null && packages.Any())
+                {
+                    errors.Add(ValidationErrors.AssignmentIsActiveInOneOrMoreDelegations, "$QUERY/cascade", [new("packages", string.Join(",", packages.Select(p => p.Id.ToString())))]);
+                }
+
+                var delegations = await delegationRepository.Get(f => f.FromId, fromEntityId, cancellationToken: cancellationToken);
```

Comment:
Delegations oppslaget her blir vel alt for bredt. Det er fortsatt bare snakk om delegations knytt til existingAssignment.Id som er relevante og vil være de som blir slettet ved cascade.

Reply:
f.FromId er vel også nettopp AssignmentId ikke party/entitity id referanse. Så er vel bare å endre til:

`var delegations = await delegationRepository.Get(f => f.FromId, existingAssignment.Id, cancellationToken: cancellationToken);`

Reply:
hehehe, skarpt øye. my bad

Reply:
fiksa nu